lib/ansible/modules/cloud/azure/azure_rm_image.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+author: Emma Laurijssens van Engelenhoven
+description:
+  - Capture an Azure Virtual Machine image for the deployment of other VMs in Azure
+     The VM should be generalized using sysprep. After this command runs, the Virtual Machine will be unusable.
+     Depends on pip azure module 2.0.0 or above and azure-mgmt-compute 2.1.0 and above.
+module: azure_rm_image
+options:
+  resource_group_name:
+    description:
+      - The resource group name to use or create to host the deployed template
+    required: true
+  name:
+    description:
+      - The name of the image being created.
+    required: True
+  vm_name:
+    description:
+      - The name of the VM within the resource group to capture. Required when state=present.
+    required: false
+  location:
+    description:
+      - The location where the image should be stored. Should be within the resource group.
+    default: The location of the resource group
+    required: false
+  state:
+    description:
+      - The desired state of the image
+    default: present
+    choices: [present, absent]
+    required: false
+  tags:
+    description:
+        - Tags to assign to the image.
+    required: false
+
+short_description: Capture Azure Virtual Machine Images
+version_added: "2.5"
+
+'''
+
+EXAMPLES = '''
+- name: Capture image
```

Comment:
Usually the examples are just tasks and not a playbook. It would be good to use actual string and not jinja2 block for some of the module parameters like `vm_name`, `name`, and so on